packages/nextjs/pages/index.tsx

```diff
@@ -3,51 +3,108 @@ import Image from "next/image";
 import Link from "next/link";
 import { useRouter } from "next/router";
 import type { NextPage } from "next";
-import { isAddress } from "viem";
+import { Address, createPublicClient, http, isAddress } from "viem";
+import * as chains from "viem/chains";
 import { MetaHeader } from "~~/components/MetaHeader";
 import { MiniFooter } from "~~/components/MiniFooter";
 import { AddressInput, InputBase } from "~~/components/scaffold-eth";
-import { fetchContractABIFromEtherscan } from "~~/utils/abi";
+import { useAbiNinjaState } from "~~/services/store/store";
+import { fetchContractABIFromEtherscan, getNetworksWithEtherscanApi, parseAndCorrectJSON } from "~~/utils/abi";
+import { notification } from "~~/utils/scaffold-eth";
+
+export const publicClient = createPublicClient({
+  chain: chains.mainnet,
+  transport: http(),
+});
```

Comment:
@portdeveloper maybe we should use `usePublicClient({chainId : parseInt(network)})` instead because we want to check on selected chain if its contract or not currently we are always checking on mainnet irrespective of network selected

Reply:
Oh thanks 🫡